c/stcbackground.c

```diff
@@ -0,0 +1,73 @@
+#include "logging.h"
+#include "stcbackground.h"
+
+static int processStcBackgroundHandler(STCBase *base, STCModule *module, int selectStatus) {
+  BackgroundModuleData* moduleData = (BackgroundModuleData*)(module->data);
+  BackgroundTask* taskList = moduleData->taskList;
+
+  for (int i = 1; i < N_TASK_TABLE_ENTRIES; i++) {
+    BackgroundTask* t = &taskList[i];
+    if (t->id != 0)  /* initialise */ {
+      t->countInterval+=STC_BACKGROUND_INTERVAL;
+      if (t->countInterval == 0) {
```

Comment:
* What if `countInterval` is not a multiple of `STC_BACKGROUND_INTERVAL`? Wouldn't you end up never calling the callback?

Let's say your interval is 17 and `STC_BACKGROUND_INTERVAL` is 9. According to the logic you'll get the following values:
|Invocation #|countInterval|
| -------------|-----:|
|0|-17|
|1|-8|
|2|1|
|3|10|
|4|19|

* What if `countInterval` is smaller than `STC_BACKGROUND_INTERVAL`? I don't think the callback will be invoked either.

Reply:
you are correct here, there are no check around it, just an assumption that it should always be equal to or greater and is a multiple of STC_BACKGROUND_INTERVAL.
Let me see what error handling i can put around this

Reply:
changed to if (t->countInterval <= 0) instead, if its negative or turned negative it will execute